### PR TITLE
fix(grpc): change bytes type to hex string

### LINKED ISSUE #377: Swagger doesn't accept hex
## Description

In order to use Swagger (OpenAPI) users should enter parameters as base64 format. Since data like block hash or transaction IDs are in hex (base16) format, this is troublesome.

### How To Reproduce

Check swagger APIs and try to get a block by hash. 

### What Happened

Input data should be in base64 format.

### Expected Behavior

It should accept base16 (hex) format for binary data.

### PR DESCRIPTION
**Pull request by Ja7ad**
## Description

Change bytes type in grpc to hex string

## Related issue(s)

- Fixes #377

### CODE REVIEW

File: crypto/bls/signature.go [hunk 1]
```diff
@@ -42,11 +42,26 @@ func SignatureFromBytes(data []byte) (*Signature, error) {
 	return &Signature{data: data}, nil
 }
 
+// SignatureFromHex return Signature from hex.
+func SignatureFromHex(sigHex string) (*Signature, error) {
```
Comment (b00f) [inline crypto/bls/signature.go:46]: Please rename it `SignatureFromString`

File: crypto/bls/signature.go [hunk 2]
```diff
@@ -42,11 +42,26 @@ func SignatureFromBytes(data []byte) (*Signature, error) {
 	return &Signature{data: data}, nil
 }
 
+// SignatureFromHex return Signature from hex.
```
Comment (b00f) [inline crypto/bls/signature.go:45]:
```suggestion
// SignatureFromString decodes the input string and returns the Signature
// if the string is a valid hex encoding of a BLS signature.
```

File: crypto/bls/signature.go [hunk 3]
```diff
@@ -42,11 +42,26 @@ func SignatureFromBytes(data []byte) (*Signature, error) {
...
+	return SignatureFromBytes(b)
+}
+
 // Bytes returns the raw byte representation of the signature.
 func (sig *Signature) Bytes() []byte {
 	return sig.data
 }
 
+// Hex return hex of signature.
+func (sig *Signature) Hex() string {
```
Comment (b00f) [inline crypto/bls/signature.go:61]: Remove this function. It is same as `String` method

File: crypto/hash/hash.go [hunk 4]
```diff
@@ -56,6 +56,15 @@ func FromBytes(data []byte) (Hash, error) {
 	return h, nil
 }
 
+func FromHex(str string) (Hash, error) {
```
Comment (b00f) [inline crypto/hash/hash.go:59]: Remove this function. It is same as `FromString`.

File: crypto/hash/hash.go [hunk 5]
```diff
@@ -70,6 +79,10 @@ func (h Hash) Bytes() []byte {
 	return h[:]
 }
 
+func (h Hash) Hex() string {
```
Comment (b00f) [inline crypto/hash/hash.go:82]: Remove this function. It is same as `String`

File: types/tx/tx.go [hunk 6]
```diff
@@ -70,6 +71,16 @@ func FromBytes(bs []byte) (*Tx, error) {
 	return tx, nil
 }
 
+// FromHex return Tx from hex.
+func FromHex(str string) (*Tx, error) {
```
Comment (b00f) [inline types/tx/tx.go:75]: Please rename it to `FromString`

File: types/tx/tx.go [hunk 7]
```diff
@@ -70,6 +71,16 @@ func FromBytes(bs []byte) (*Tx, error) {
 	return tx, nil
 }
 
+// FromHex return Tx from hex.
```
Comment (b00f) [inline types/tx/tx.go:74]: // FromString constructs a new transaction from a hex-encoded string.

File: types/tx/tx.go [hunk 8]
```diff
@@ -221,6 +232,16 @@ func (tx *Tx) Bytes() ([]byte, error) {
 	return w.Bytes(), nil
 }
 
+// Hex return hex of Tx.
+func (tx *Tx) Hex() (string, error) {
```
Comment (b00f) [inline types/tx/tx.go:236]: Please rename it to `String` . On error it returns ""

**Review by b00f — CHANGES_REQUESTED** [review]

File: crypto/hash/hash_test.go [hunk 9]
```diff
@@ -29,6 +29,14 @@ func TestHashFromString(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestHashFromBytes(t *testing.T) {
```
Comment (b00f) [inline crypto/hash/hash_test.go:32]: Nothing has changed in the implementation, so this test is not necessary.

File: node/peers.json [hunk 10]
```diff
@@ -0,0 +1 @@
...,"/ip4/42.117.247.180/tcp/21888/p2p/12D3KooWLVZSkgkFgrRNMfkCtULqtErYcNgjqgueeH944vDpDCPv","/ip4/65.109.143.78/tcp/21888/p2p/12D3KooWPnWgEFV32cvpfuv98oioZQJJKvRM46BEsFaoXYijvgvc","/dns/pactus-bootstrap.stakers.world/tcp/21888/p2p/12D3KooWK1DAcxf8RJA2PVkwr9ehk1zgXcnjLyPCsYCpvw1NSfhc","/ip4/95.216.222.135/tcp/21888/p2p/12D3KooWHicbZ3dBnR7JaVKr4U7WNUWyhDnUn2TwMVmPqW8Ftqop","/ip4/43.135.166.197/tcp/21888/p2p/12D3KooWEqyjMNzFe92HFZMZQd3YZ6zvztuPaC3a4wAUvPf4moyP","/dns/bootstrap3.pactus.org/tcp/21888/p2p/12D3KooWBCPSZWheet6tMoHbVBCDfBwQm4yzCwcQ8hJ6NMCN97sj","/dns/bootstrap2.pactus.org/tcp/21888/p2p/12D3KooWM39ag7ghta49qybf7McADgT8FLakTYkCsiBvwdnjuG5q","/ip4/62.171.130.196/tcp/21888/p2p/12D3KooWSkwB8AiWz3VGyG5cgC3ry5dmCw3TthQNc1jvRNe9EEKH","/ip4/49.13.27.37/tcp/21888/p2p/12D3KooWPSh9KMa6tRfSewK3sUC4BDSPnXseVDGJxWqNZxKcXnGM","/ip4/37.60.245.47/tcp/21888/p2p/12D3KooWBH27VEPdGBYzq1vYciRgnDn3ozStRaUYd2vJWGgaKEUZ","/dns/bootstrap1.pactus.org/tcp/21888/p2p/12D3KooWMnDsu8TDTk2VV8uD8zsNSB6eUkqtQs6ttg4bHq9zNaBe","/ip4/46.250.233.5/tcp/21888/p2p/12D3KooWH1kam9g1QFWmtWMLzP2MDY4ptpLTMPU8wSq3Q4k3rvnn","/dns/pactus-bootstrap1.dezh.tech/tcp/21888/p2p/12D3KooWK1z7QAskVrQd98r98UMSPwTLr4out8B9NkQTrCdZPCZx"]
```
Comment (b00f) [inline node/peers.json:1]: Revert this file and avoid running Pactus on workspace.

**Review by b00f — CHANGES_REQUESTED** [review]

**Comment by b00f** [comment]
@Ja7ad 

In `tx` package you added two methods: `FromHex` and `Hex`

The `Hex` is never used in implementation, `FromHex` is used only once.
Since `tx` is a big object, it make sense that we don't add From/To string to it.

So, remove these function. 
Inside tests you can use: `ts.DecodingHex` for decoding hex easily.

**Comment by Ja7ad** [comment]
@b00f done

**Review by b00f — APPROVED** [review]

**Pull request merged**